### PR TITLE
Fixed typos in deep dive docs

### PR DESCRIPTION
**Pull request by lucasalavapena**
I also added the missing function name (`issubclass`) in the arrays code example, otherwise the check is a bit odd in my opinion.

### CODE REVIEW

File: docs/overview/deep_dive/ivy_tests.rst [hunk 1]
```diff
@@ -354,7 +354,7 @@ The test then runs on the newly created arrays with specified data types.
 Why do we need helper functions
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-It is usually the case that any ivy function should run seamlessly on ‘all the possible varieties, as well as the edge cases’ encountered by the following parameters -:
+It is usually the case that any ivy function should run seamlessly on `all the possible varieties, as well as the edge cases` encountered by the following parameters -:
```
Comment (KareemMAX) [inline docs/overview/deep_dive/ivy_tests.rst:357]: Why do you think that is better?

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hey @lucasalavapena,
Great job, made some few comments

File: docs/overview/deep_dive/ivy_tests.rst (hunk 1 above)
Comment (lucasalavapena) [reply docs/overview/deep_dive/ivy_tests.rst:357]: I thought that is what they were going for, and it was an accident that they did not italise it. What are your thoughts?

Comment (KareemMAX) [reply docs/overview/deep_dive/ivy_tests.rst:357]: I guess they were going for `'`, but they may have used a word processor that stylized the quotes. I like the stylized version more and see no need to remove it

**Review by KareemMAX — APPROVED** [review]

**Pull request merged**